spec/system/dashboards/user_sorts_dashboard_articles_spec.rb

```diff
@@ -1,8 +1,6 @@
 require "rails_helper"
 
-RSpec.describe "Sorting Dashboard Articles", type: :system, js: true, db_strategy: :truncation do
-  self.use_transactional_tests = false
-
+RSpec.describe "Sorting Dashboard Articles", type: :system, js: true do
```

Comment:
@djuber Does this spec actually fire off a headless-browser-based test, via `js: true`? I remember with PhantomJS (via the `poltergeist` gem and other headless Capybara plugins), it wasn't possible to use transactions with those particular kinds of tests because the code making the changes to the DB wasn't being run inside a transaction — it was being run inside a forked process.

If this is still the case, we may need to keep truncation. But also I haven't been doing a lot of work with back-end tests dealing with JS for the past few years and I don't know whether that landscape has changed.

Reply:
https://github.com/forem/forem/blob/master/docs/tests/test-flags.md#js-true-flag covers what we _think_ this is doing

https://github.com/forem/forem/blob/master/spec/support/initializers/capybara.rb#L14-L28 looks like the place the handler is defined - either use chrome headless or remote chrome - the poltergeist/forked process issues hopefully do not apply here (rspec is driving chrome but didn't need to fork?)

We definitely have many more cases of system tests setting js: true than we had db cleaner truncation strategy tests (there were only the two being handled) so I would be surprised if the truncation / dbcleaner setup were needed generally and only added in two cases.

Reply:
@djuber I just grepped all the gems we use for `fork` . `selenium-webkit` uses the `childprocess` gem which [does call `fork`](https://github.com/enkessler/childprocess/blob/v3.0.0/lib/childprocess/unix/fork_exec_process.rb#L20), but I don't know if that's actually invoked. It's a downstream dependency of [the `webdrivers` gem](https://github.com/forem/forem/blob/195b202d851578800b993e2d5312dfbce49f8e8e/Gemfile#L176), but that loads several different ones, so I'm not completely sure.